matplotlib2tikz/legend.py

```diff
@@ -106,44 +107,74 @@ def draw_legend(data, obj):
 
     if loc == 1:
         # upper right
-        position = None
+        if obj._bbox_to_anchor:
```

Comment:
Yes you are right it is not very clean, but I have already tried it before, but didn't found an easier way that is even shorter.  I guess with one big if-else it would be twice as long as now.